cmd/starter/main.go

```diff
@@ -130,6 +131,26 @@ func fetchBuildJob() {
 	}
 	repo := job.Repository
 
+	// Aquire lock to ensure one repo is not built twice
+	lockID := uuid.NewV4().String()
+	redisClient.Set(fmt.Sprintf("project::%s::build-lock", repo), lockID, 1800, 0, false, true)
+
+	lock, err := redisClient.Get(fmt.Sprintf("project::%s::build-lock", repo))
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"error": err,
+		}).Error("Unable to fetch lock state")
+		return
+	}
+
+	if string(lock) != lockID {
+		// If build-job is currently locked but not by us, try again later
+		queueEntry, err := job.ToByte()
+		orFail(err)
+		redisClient.RPush("build-queue", string(queueEntry))
+		return
+	}
```

Comment:
I don't really know. The code for the set command is quite intransparent… Thats why I wrote the get check to ensure the own ID is set to the lock. Any other sets should fail and the id does not match so the get returns from the function…